domain/src/main/java/com/globant/domain/usecase/implementation/GetAllCharactersUseCaseImpl.kt

```diff
@@ -1,10 +1,18 @@
 package com.globant.domain.usecase.implementation
 
 import com.globant.domain.entity.Character
-import com.globant.domain.repository.CharacterRepository
+import com.globant.domain.service.CharactersService
 import com.globant.domain.usecase.GetAllCharactersUseCase
+import com.globant.domain.util.Constants
 import com.globant.domain.util.Result
 
-class GetAllCharactersUseCaseImpl(private val characterRepository: CharacterRepository): GetAllCharactersUseCase {
-    override fun invoke(): Result<List<Character>> = characterRepository.getAllCharacters()
+class GetAllCharactersUseCaseImpl(private val characterService: CharactersService) : GetAllCharactersUseCase {
+    override fun invoke(): Result<List<Character>> {
+        characterService.getAllCharacters().let {
```

Comment:
the when must be in the serviceImpl

here you will just do something like

override fun invoke(): Result<List<Character>> = characterService.getAllCharacters()

Reply:
refactored, thanks